### PR TITLE
Add support for rate limit exceeded exception

### PR DESCRIPTION
**Pull request by piodul**
This PR adds support for a new Scylla-specific type of error returned from the database: RateLimitExceeded. It was introduced as a part of the _per-partition rate limit_ feature, upcoming in Scylla 5.1 (PR in Scylla: https://github.com/scylladb/scylladb/pull/9810). The RateLimitExceeded error is sent when Scylla detects that the per-partition rate limit is exceeded.

In order for Scylla to be able to send this error, the driver must tell the database during connection handshake that it is able to interpret it (otherwise Scylla falls back to the Config_error error). This PR adds support for Scylla-specific protocol extensions, negotiates the `SCYLLA_RATE_LIMIT_ERROR` during handshake and adds support for the RateLimitExceeded error deserialization.

## Pre-review checklist

- [x] I have split my patch into logically separate commits.
- [x] All commit messages clearly explain what they change and why.
- [x] I added relevant tests for new features and bug fixes.
- [x] All commits compile, pass static checks and pass test.
- [x] PR description sums up the changes and reasons why they should be introduced.
- [ ] ~I added appropriate `Fixes:` annotations to PR description.~

### CODE REVIEW

**Comment by psarna** [comment]
I assume it was tested manually - if so, please add a note in the PR description

**Comment by psarna** [comment]
Oh wait, there are automatic tests too, nvm

**Comment by psarna** [comment]
Nope, still valid - please let me know if you saw the new code successfully set up a connection to new Scylla and negotiated the new error type (e.g. via wireshark or logs)

**Review by psarna — APPROVED** [review]
looks good to me, waiting for confirmation about manual tests

**Comment by piodul** [comment]
I added an automatic test. It automatically detects support for per-partition rate limiting and skips itself if not supported. The CI will be skipping this test until Scylla 5.1 is released and we update the version we use in CI.

I ran the test with Scylla master and found out that it has a bug and doesn't send the new error even if it is properly negotiated (https://github.com/scylladb/scylladb/issues/11517). The fix is an one-liner and after applying it the test passes. I'll send the fix shortly.

**Comment by piodul** [comment]
I sent the aforementioned fix to Scylla last friday, forgot to provide a link to it here: https://github.com/scylladb/scylladb/pull/11518

**Comment by psarna** [comment]
Fix for ScyllaDB is queued, let's wait until it's promoted upstream and then I'll merge this one

**Pull request merged**
